store/kv/chunk.go

```diff
@@ -0,0 +1,235 @@
+				return err
+			}
+		} else {
+			// we don't care about any other attributes for this case
+			chunked := internal.NewTableData(chunkData)
+			// int is encoded as int64 by FDB client so better use int64 to avoid any confusion.
+			chunkKey = append(chunkKey, []KeyPart{chunkIdentifier, int64(chunk)}...)
+
+			if err := tx.KeyValueTx.Replace(ctx, table, chunkKey, chunked, false); err != nil {
+				return err
+			}
+		}
+
+		return nil
+	})
+}
+
+func (tx *ChunkTx) Replace(ctx context.Context, table []byte, key Key, data *internal.TableData, isUpdate bool) error {
+	if !tx.isChunkingNeeded(data) {
+		return tx.KeyValueTx.Replace(ctx, table, key, data, isUpdate)
+	}
+
+	tx.setTotalChunks(data)
+
+	// cleanup the existing range
+	if err := tx.KeyValueTx.Delete(ctx, table, key); err != nil {
+		return err
+	}
+
+	return tx.executeChunks(data, func(chunk int32, chunkData []byte) error {
+		var chunked *internal.TableData
+		chunkKey := key
+		if chunk == 0 {
+			// first part
+			chunked = data.CloneWithAttributesOnly(chunkData)
+		} else {
+			chunked = internal.NewTableData(chunkData)
+			// int is encoded as int64 by FDB client so better use int64 to avoid any confusion.
+			chunkKey = append(chunkKey, []KeyPart{chunkIdentifier, int64(chunk)}...)
+		}
+
+		if err := tx.KeyValueTx.Replace(ctx, table, chunkKey, chunked, isUpdate); err != nil {
+			return err
+		}
+
+		return nil
+	})
+}
+
+// Read needs to return chunk iterator so that it can merge and returned merged chunk to caller.
+func (tx *ChunkTx) Read(ctx context.Context, table []byte, key Key) (Iterator, error) {
+	iterator, err := tx.KeyValueTx.Read(ctx, table, key)
+	if err != nil {
+		return nil, err
+	}
+
+	return &ChunkIterator{
+		Iterator: iterator,
+	}, nil
+}
+
+func (tx *ChunkTx) ReadRange(ctx context.Context, table []byte, lKey Key, rKey Key, isSnapshot bool) (Iterator, error) {
+	iterator, err := tx.KeyValueTx.ReadRange(ctx, table, lKey, rKey, isSnapshot)
+	if err != nil {
+		return nil, err
+	}
+
+	return &ChunkIterator{
+		Iterator: iterator,
+	}, nil
+}
+
+type ChunkIterator struct {
+	Iterator
+
+	err error
+}
+
+func (it *ChunkIterator) Next(value *KeyValue) bool {
+	if !it.Iterator.Next(value) {
+		return false
+	}
+
+	if !value.Data.IsChunkedData() {
+		return true
+	}
+
+	var buf bytes.Buffer
+	buf.Write(value.Data.RawData)
+	hasNext := false
+	for chunk := int32(1); chunk < *value.Data.TotalChunks; chunk++ {
+		var chunked KeyValue
+		if hasNext = it.Iterator.Next(&chunked); !hasNext {
+			break
+		}
+		if len(chunked.Key) < 2 {
+			it.err = fmt.Errorf("key shorter than expected chunked key")
+			return false
+		}
+
+		if chunked.Key[len(chunked.Key)-2] != chunkIdentifier {
+			it.err = fmt.Errorf("no chunk number found")
+			return false
+		}
+
+		if chunkNo, ok := chunked.Key[len(chunked.Key)-1].(int64); !ok || chunk != int32(chunkNo) {
+			it.err = fmt.Errorf("chunk number mismatch")
+			return false
+		}
+
+		buf.Write(chunked.Data.RawData)
+	}
+
```

Comment:
You can check that number of chunk read is equal to the number of chunk in the first header.